mlrun/runtimes/nuclio/application/application.py

```diff
@@ -883,3 +992,60 @@ def _sync_api_gateway(self):
         self.status.api_gateway = APIGateway.from_scheme(api_gateway_scheme)
         self.status.api_gateway.wait_for_readiness()
         self.url = self.status.api_gateway.invoke_url
+
+    def _enrich_sidecar_probe_ports(self):
+        """Enrich sidecar probe configurations with internal application port if needed
+
+        This method is called just before deployment to automatically enrich HTTP probes
+        in the sidecar container that were configured without an explicit port.
+
+        Enrichment logic:
+        - Only enriches HTTP probes (httpGet) that don't already have a port specified
+        - If the user explicitly provided http_port in set_probe(), enrichment is skipped
+        - If the user provided a port in the config dict, enrichment is skipped
+        - Enrichment happens just before deployment to ensure the latest internal_application_port
+          value is used, even if it was modified after set_probe() was called
+        """
+        internal_port = None
+        if hasattr(self.spec, "internal_application_port"):
+            internal_port = self.spec.internal_application_port
+
+        # Check each probe type and enrich missing HTTP ports
+        sidecar = self._get_sidecar()
+        if not sidecar:
+            return
+
+        for probe_type in ProbeType:
+            probe_key = PROBE_KEYS[probe_type.value]
+            probe_config = sidecar.get(probe_key)
+
+            if probe_config and isinstance(probe_config, dict):
+                http_get = probe_config.get("httpGet")
+                if http_get and isinstance(http_get, dict) and "port" not in http_get:
+                    if internal_port is None:
+                        raise ValueError(
+                            f"Cannot enrich {probe_type.value} probe: HTTP probe requires a port, "
+                            "but internal_application_port is not set. "
+                            "Please set the internal_application_port or provide http_port in set_probe()."
```

Comment:
just fyi - technically it should be none, because we always enrich it here https://github.com/mlrun/mlrun/blob/a3f2f2f6678de5a462ffa65be0ec2ee295bede73/mlrun/runtimes/nuclio/application/application.py#L162-L165 . So it can be none only if user explicitly set `_internal_application_port` to None

Reply:
@weilerN not really